pkg/restapi/vc/operation/operations.go

```diff
@@ -849,50 +881,49 @@ func (o *Operation) createHolderProfile(pr *HolderProfileRequest) (*vcprofile.Ho
 		SignatureType:           pr.SignatureType,
 		SignatureRepresentation: pr.SignatureRepresentation,
 		Creator:                 publicKeyID,
-		DIDPrivateKey:           didPrivateKey,
-		DIDKeyType:              pr.DIDKeyType,
 		OverwriteHolder:         pr.OverwriteHolder,
 	}, nil
 }
 
-func (o *Operation) createProfile(keyType, signatureType, did, didPrivateKey, purpose string,
-	registrar UNIRegistrar) (string, string, string, error) {
+func (o *Operation) createProfile(keyType, signatureType, did, privateKey, keyID, purpose string,
+	registrar UNIRegistrar) (string, string, error) {
 	var didID string
 
 	var publicKeyID string
 
 	switch {
 	case registrar.DriverURL != "":
 		var err error
-		didID, publicKeyID, didPrivateKey, err = o.createDIDUniRegistrar(keyType, signatureType, purpose, registrar)
+		didID, publicKeyID, err = o.createDIDUniRegistrar(keyType, signatureType, purpose, registrar)
 
 		if err != nil {
-			return "", "", "", err
+			return "", "", err
 		}
 
 	case did == "":
 		var err error
 		didID, publicKeyID, err = o.createDID(keyType, signatureType)
 
 		if err != nil {
-			return "", "", "", err
+			return "", "", err
 		}
 
 	case did != "":
 		didDoc, err := o.vdri.Resolve(did)
 		if err != nil {
-			return "", "", "", fmt.Errorf("failed to resolve did: %v", err)
+			return "", "", fmt.Errorf("failed to resolve did: %v", err)
 		}
 
 		didID = didDoc.ID
 
-		publicKeyID, err = getPublicKeyID(didDoc, "", crypto.Ed25519Signature2018)
-		if err != nil {
-			return "", "", "", err
+		if err := o.importKey(keyID, kms.ED25519Type, base58.Decode(privateKey)); err != nil {
```

Comment:
Are we assuming that if did is there that key has to be Ed25519 type? This can be handled in separate PR - it is an existing issue.

Reply:
for now yes because all vendors now just support ed25519

Reply:
need to change when vendor start support multiple keys